awspub/notification.py

```diff
@@ -0,0 +1,94 @@
+"""
+Methods used to handle notifications for AWS using SNS
+"""
+
+import json
+import logging
+from typing import Any, Dict
+
+import boto3
+from mypy_boto3_sns.client import SNSClient
+
+from awspub.exceptions import AWSNotificationException
+
+logger = logging.getLogger(__name__)
+
+
+class SNSNotification(object):
+    """
+    A data object that contains validation logic and
+    structuring rules for SNS notification JSON
+    """
+
+    def __init__(self, conf: Dict[str, Any], image_id: str, region_name: str):
+        """
+        Construct a message and verify that it is valid
+        """
+        self._conf = conf
+        self._image_id = image_id
+        self._region_name = region_name
+
+    def _get_topic(self, client: SNSClient) -> str:
+        """
+        Get topic ARN
```

Comment:
I think it would be possible to calculate the ARN without iterating over the existing topics. Format is:
```
arn:partition:service:region:account-id:resource-id
```
See https://docs.aws.amazon.com/IAM/latest/UserGuide/reference-arns.html . 

What do you think about that? I'm not sure what's better but I kind of like the fact that we don't have to query the API for this...

Reply:
changed with your suggestion.